src/app/pages/sharing/iscsi/target/target-form/target-form.component.ts

```diff
@@ -1,255 +1,182 @@
-            initiatorGroupField.options.push({ label: optionLabel, value: initiator.id });
-          });
-          resolve(true);
-        },
-        () => {
-          resolve(false);
-        },
-      );
-    });
-    const promise3 = new Promise((resolve) => {
-      this.iscsiService.getAuth().toPromise().then(
-        (accessRecords) => {
-          const tags = _.uniq(accessRecords.map((item) => item.tag));
-          authGroupField.options.push({ label: 'None', value: null });
-          for (const tag of tags) {
-            authGroupField.options.push({ label: String(tag), value: tag });
-          }
-          resolve(true);
-        },
-        () => {
-          resolve(false);
-        },
-      );
+  setTargetForEdit(target: IscsiTarget): void {
+    this.editingTarget = target;
+
+    Object.values(target.groups).forEach((group, index) => {
+      this.groupsFromControls.forEach((fc) => {
+        this.form.addControl(`${fc.name}${this.listPrefix}${index}`, new FormControl(fc.name, fc?.validator));
+      });
+      this.groups.push(group);
     });
 
-    return Promise.all([promise1, promise2, promise3]).then(
-      () => true,
-    );
+    this.form.patchValue({
+      ...target,
+    });
   }
 
-  afterInit(entityForm: EntityFormComponent): void {
-    this.entityForm = entityForm;
-    this.fieldConfig = entityForm.fieldConfig;
-    this.title = entityForm.isNew ? this.translate.instant('Add ISCSI Target') : this.translate.instant('Edit ISCSI Target');
+  onSubmit(): void {
+    const values = this.form.value;
+
+    const params = {
+      name: values.name,
+      alias: values.alias,
+      mode: values.mode,
+      groups: this.groups,
+    };
+
+    this.isLoading = true;
+    let request$: Observable<unknown>;
+    if (this.isNew) {
+      request$ = this.ws.call('iscsi.target.create', [params]);
+    } else {
+      request$ = this.ws.call('iscsi.target.update', [
+        this.editingTarget.id,
+        params,
+      ]);
+    }
+
+    request$.pipe(untilDestroyed(this)).subscribe(() => {
+      this.isLoading = false;
+      this.slideInService.close();
+    }, (error) => {
+      this.isLoading = false;
+      this.errorHandler.handleWsFormError(error, this.form);
+      this.cdr.markForCheck();
+    });
   }
 
-  customEditCall(value: IscsiTargetUpdate): void {
-    this.loader.open();
-    this.ws.call(this.editCall, [this.pk, value]).pipe(untilDestroyed(this)).subscribe(
-      () => {
-        this.loader.close();
-        this.modalService.closeSlideIn();
-      },
-      (res) => {
-        this.loader.close();
-        new EntityUtils().handleWsError(this.entityForm, res);
-      },
-    );
+  addGroup(): void {
+    const newIndex = this.groups.length;
+    const newListItem: any = {};
+    this.groupsFromControls.forEach((fc) => {
+      newListItem[fc.name] = fc.default;
+      this.form.addControl(`${fc.name}${this.listPrefix}${newIndex}`, new FormControl(fc.default, fc?.validator));
```

Comment:
The `?` in `fc?.validator` doesn't really do anything since we're already accessing `fc` without checking in `fc.default`. Better check should be used if the purpose is to check if `fc.validator` exists. Same in another place above.

Reply:
> If this issue is out of scope for this ticket
Yes, I suppose this is out of scope. I guess it's also needs fixing in other places where `ix-list` is used. Let's create a separate ticket

Reply:
>fc?.validator doesn't really do anything

It wasn't intentional, just a typo. Fixed now!